.github/workflows/risc0-ethereum.yml

```diff
@@ -0,0 +1,157 @@
+name: risc0-ethereum CI
+
+on:
+  merge_group:
+  pull_request:
+    branches: [main, "release-*"]
+  workflow_dispatch:
+    inputs:
+      risc0-ethereum-branch:
+        description: "risc0-ethereum branch to checkout"
+        required: false
+        default: 'main'
+      bonsai-foundry-template-branch:
+        description: "bonsai-foundry-template branch to checkout"
+        required: false
+        default: 'main'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+# this is needed to gain access via OIDC to the S3 bucket for caching
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  RISC0_TOOLCHAIN_VERSION: v2024-02-08.1
+  RISC0_PATH: ${{ github.workspace }}/risc0
+  RISC0_ETHEREUM_PATH: ${{ github.workspace }}/risc0-ethereum
+  BONSAI_FOUNDRY_TEMPLATE_PATH: ${{ github.workspace }}/bonsai-foundry-template
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      test: ${{ steps.filter.outputs.test }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: risc0/paths-filter@4067d885736b84de7c414f582ac45897079b0a78
+        id: filter
+        with:
+          filters: |
+            base: &base
+              - .github/actions/**
+              - .github/workflows/risc0-ethereum.yml
+              - Cargo.toml
+              - rust-toolchain.toml
+              - Cargo.lock
+            test:
+              - *base
+              - bonsai/**
+              - risc0/**
+
+  # see: https://github.com/orgs/community/discussions/26822
+  risc0-ethereum-status-check:
+    if: always()
+    needs:
+      - test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check all job status
+        # see https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#needs-context
+        # see https://stackoverflow.com/a/67532120/4907315
+        if: ${{ contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled') }}
+        run: exit 1
+
+  test:
+    if: ${{ needs.changes.outputs.test == 'true' || github.event_name == 'workflow_dispatch'}}
+    needs: changes
+    runs-on: [self-hosted, prod, Linux, cpu, docker]
+    env:
+      RUST_BACKTRACE: full
+    steps:
+      # This is a workaround from: https://github.com/actions/checkout/issues/590#issuecomment-970586842
```

Comment:
This is another comment, no action needed at this time: so now that we don't have any sub-modules in risc0, we could consider dropping this workaround. We can still keep it around though